src/components/accounts/PasswordSettings.js

```diff
@@ -0,0 +1,247 @@
+/* eslint-disable no-unused-vars */
+import React, { Component } from 'react';
+import { Button, Label, FormGroup } from 'react-bootstrap';
+
+import { validatePassword } from '../utils';
+import {
+    PASSWORD_WARNING,
+    VERIFY_PASSWORD_ERROR,
+    INCORRECT_PASSWORD_WARNING,
+    PASSWORD_CHANGE_ERROR,
+    PASSWORD_CHANGE_UNSUCCESSFUL,
+    PASSWORD_CHANGE_SUCCESSFUL
+} from '../../constants';
+import '../../css/Settings.css';
+/* eslint-enable no-unused-vars */
+
+class PasswordSettings extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            currentPassword: '',
+            newPassword: '',
+            newPasswordReentry: '',
+            currentPasswordIsValid: undefined,
+            newPasswordIsValid: undefined,
+            newPasswordReentryIsValid: undefined,
+            isPasswordChangeSuccessful: undefined,
+            passwordChangeMessage: undefined
+        };
+
+        this.validatePasswordReentry = this.validatePasswordReentry.bind(this);
+        this.handleInputChange = this.handleInputChange.bind(this);
+        this.validateCurrentPassword = this.validateCurrentPassword.bind(this);
+        this.handlePasswordChangeSubmit = this.handlePasswordChangeSubmit.bind(this);
+        this.updatePassword = this.updatePassword.bind(this);
```

Comment:
take a look at this, regarding `bind()`: https://github.com/ubclaunchpad/inertia/issues/375